Utils/AsyncTaskList.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DUCK.Utils
+{
+	public class AsyncTaskList
+	{
+		private readonly List<Action<Action>> tasks = new List<Action<Action>>();
```

Comment:
Is there a reason why a List collection was chosen over a Queue? If you are not persisting the list (aka you need to create a new List every time you want to start a task chain) then queue describes the behaviour a little bit better.

Reply:
Going back to my previous comment on using the underlying enumerator so that you don't have to maintain a state of where you are in the list and can instead use MoveNext. 

Queue would give you a similar benefit with dequeue. +1

Reply:
So I've made it a queue, which means you can't persist the list, (cannot run it more than once). We also now guard against calling run multiple times